Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -642,27 +656,48 @@ this has no time dependence prior to this operation.
 void ComponentInfo::merge(const ComponentInfo &other) {
   if (!m_scanCounts)
     this->initScanCounts();
-  const auto toMerge = buildMergeIndices(other);
-  if (m_isSyncScan)
-    throw std::runtime_error(
-        "Sync scans not currently supported for ComponentInfo");
-  initIndices();
-  auto scanCounts(m_scanCounts);
-  for (size_t linearIndex = 0; linearIndex < other.m_positions->size();
-       ++linearIndex) {
-    if (!toMerge[linearIndex])
-      continue;
-    auto newIndex = getIndex(other.m_indices, linearIndex);
-    const size_t compIndex = newIndex.first;
-    newIndex.second += scanCount(compIndex);
-    scanCounts.access()[compIndex]++;
-    m_indexMap.access()[compIndex].push_back((*m_indices).size());
-    m_indices.access().push_back(newIndex);
-    m_positions.access().push_back((*other.m_positions)[linearIndex]);
-    m_rotations.access().push_back((*other.m_rotations)[linearIndex]);
-    m_scanIntervals.access().push_back((*other.m_scanIntervals)[linearIndex]);
-  }
-  m_scanCounts = std::move(scanCounts);
+  if (m_isSyncScan) {
+    const auto &toMerge = buildMergeIndicesSync(other);
+    for (size_t timeIndex = 0; timeIndex < other.m_scanIntervals->size();
+         ++timeIndex) {
+      if (!toMerge[timeIndex])
+        continue;
+      auto &scanIntervals = m_scanIntervals.access();
+      auto &positions = m_positions.access();
+      auto &rotations = m_rotations.access();
+      m_scanCounts.access()[0]++;
+      scanIntervals.push_back((*other.m_scanIntervals)[timeIndex]);
+      const size_t indexStart = other.linearIndex({0, timeIndex});
+      size_t indexEnd = indexStart + nonDetectorSize();
+      positions.insert(positions.end(), other.m_positions->begin() + indexStart,
+                       other.m_positions->begin() + indexEnd);
+      rotations.insert(rotations.end(), other.m_rotations->begin() + indexStart,
+                       other.m_rotations->begin() + indexEnd);
+    }
+
+  } else {
+    const auto toMerge = buildMergeIndices(other);
+    if (m_isSyncScan)
```

Comment:
Something is in the wrong order here. This in within the `else` block of `if (m_isSyncScan)`.

But I suppose it will need to change anyway, based on our discussion on Slack yesterday.